src/main/scala/org/apache/spark/sql/pulsar/PulsarWriteTask.scala

```diff
@@ -209,6 +209,11 @@ private[pulsar] abstract class PulsarRowWriter(
 
   protected def producerClose(): Unit = {
     producerFlush()
+    if (singleProducer != null) {
+      singleProducer.close()
+    } else {
+      topic2Producer.foreach(_._2.close())
```

Comment:
Do we need to handle exception here?

Reply:
If the exception happens, we will let it propagate. 

And it will be catched here: https://github.com/streamnative/pulsar-spark/blob/master/src/main/scala/org/apache/spark/sql/pulsar/PulsarSinks.scala#L153